lib/internal/modules/esm/fetch_module.js

```diff
@@ -263,15 +262,9 @@ function fetchModule(parsed, { parentURL }) {
   if (parsed.protocol === 'http:') {
     return PromisePrototypeThen(isLocalAddress(parsed.hostname), (is) => {
       if (is !== true) {
-        let parent = parentURL;
-        const parentName = path.basename(parent.pathname);
-        if (
-          parentName === '[eval]' ||
-          parentName === '[stdin]'
-        ) parent = 'command-line';
         throw new ERR_NETWORK_IMPORT_DISALLOWED(
           href,
-          parent,
+          parentURL,
```

Comment:
The reason I added this was because the output without it seemed likely to confuse. Aside from consistency with CJS, I think this is not an improvement (I don't feel strongly though).

Reply:
Note that the condition is never true (`parentName` is never equals to `[eval]` not `[stdin]`, it could be `[eval1]` though). If the URL is confusing, we should fix the URL itself, not try to hide it in this specific error message imo.

Reply:
> If the URL is confusing, we should fix the URL itself, not try to hide it in this specific error message imo.

Sure! That works for me (and should also better support consistency).

> what if the user didn't start the process via the command-line?

Is that possible?

> Note that the condition can never be true: parentName is never equals to [eval] nor [stdin] (it could be [eval1] though, but we're not testing that).

If it doesn't now, it did at the time—I specifically tested CLI via `-e …`, which is how I found this in the first place 🙂 Is there a known related change that would explain why it's not now?

Reply:
> > Is that possible?
> 
> 
> 
> Sure, why not? I don't have a deep enough knowledge of the various OSs Node.js supports to list them all, but I know that CLI is just an option among others (from the GUI, as a service, from a parent process, etc.).

No, I mean, is there a route that exists that leads to `[eval*]`/`[stdin]` that did not originate from CLI?

> > If it doesn't now, it did at the time—I specifically tested CLI via `-e …`, which is how I found this in the first place 🙂 Is there a known related change that would explain why it's not now?
> 
> 
> 
> Maybe you forgot to add `--input-type=module`? 

I did not 🙂